pkg/client2/client.go

```diff
@@ -0,0 +1,395 @@
+package client2
+
+import (
+	"context"
+	"crypto/ed25519"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/CoreumFoundation/coreum-tools/pkg/retry"
+	"github.com/cosmos/cosmos-sdk/client"
+	cosmosed25519 "github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	cosmossecp256k1 "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	cosmoserrors "github.com/cosmos/cosmos-sdk/types/errors"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/pkg/errors"
+	coretypes "github.com/tendermint/tendermint/rpc/core/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/CoreumFoundation/coreum/app"
+	"github.com/CoreumFoundation/coreum/pkg/tx2"
+	"github.com/CoreumFoundation/coreum/pkg/types"
+)
+
+const (
+	requestTimeout       = 10 * time.Second
+	txTimeout            = time.Minute
+	txStatusPollInterval = 500 * time.Millisecond
```

Comment:
Those params should be configarable for the client, so you should be able to pass the "Config" with the them to the "New" func. Or use the DefaultConfig() with the defaults.